stl/inc/memory

```diff
@@ -846,9 +850,7 @@ protected:
     template <class _Ty2>
     void _Copy_construct_from(const shared_ptr<_Ty2>& _Other) noexcept {
         // implement shared_ptr's (converting) copy ctor
-        if (_Other._Rep) {
-            _Other._Rep->_Incref();
-        }
+        _Other._Incref();

```

Comment:
I introduced this _Incref function to avoid code repetition of checking for _Rep. I changed `_Copy_construct_from` and `_Alias_construct_from` to keep the code clean.